### ISSUE TITLE
Azure.AI.OpenAI: Support setting "name" property for "system" role

### ISSUE THREAD

**Issue by joakimriedel** [Azure/azure-sdk-for-net#36380]
### Library name

Azure.AI.OpenAI

### Please describe the feature.

According to [documentation on few-shot prompting for the gpt-3.5-turbo model](https://github.com/openai/openai-cookbook/blob/main/examples/How_to_format_inputs_to_ChatGPT_models.ipynb) (**In [9]** section) it might be beneficial to provide examples for the `system` role together with setting the `name` property with `example_user` and `example_assistant` values. Looking at [the ChatMessage class](https://github.com/Azure/azure-sdk-for-net/blob/Azure.AI.OpenAI_1.0.0-beta.5/sdk/openai/Azure.AI.OpenAI/src/Custom/ChatMessage.cs) it only seems to support `role` and `content` but not setting the `name` property. Please implement setting the `name` property of `ChatMessage`, or optionally on `ChatRole` when it is set to `system`.

**Comment by daxian-dbw** [comment]
I also ran into this limitation. Can we please get some traction here? Thanks!

**Comment by trrwilson** [comment]
Thank you for filing this issue and apologies for the slow turnaround as we've established our process!

Given this is a change we'll want to reflect across languages via the shared service definition, I've created a copy of this issue over in the azure-rest-api-specs project to the track the central validation and implementation. Once we're able to add support there, the next snap of the definition taken by the .NET SDK will include support. I'll close this issue in favor of the language-independent one to keep things crisp; updates will coalesce to that issue.

https://github.com/Azure/azure-rest-api-specs/issues/24867
